File: 백준/Silver/18222. 투에－모스 문자열/투에－모스 문자열.py
def solve():
    k = int(input())
    cnt = 0
    i = 1
    while i * 2 < k:
        i *= 2
    while k != 1:
        if i < k:
            cnt +=1
            k -= i
        i //= 2
    print(1 if cnt % 2 else 0)

solve()

File: 백준/Silver/18222. 투에－모스 문자열/투에－모스 문자열.java
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        long k = sc.nextLong();
        long i = 1L, cnt = 0L;
        while (i * 2 < k)
            i *= 2;
        while (k != 1) {
            if (i < k) {
                cnt++;
                k -= i;
            }
            i /= 2;
        }
        System.out.println(cnt % 2 == 0 ? 0 : 1);
    }

}